rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/clearingrequest/Sw360ClearingRequestService.java

```diff
@@ -40,10 +48,43 @@ private ModerationService.Iface getThriftModerationClient() throws TTransportExc
     }
 
     public ClearingRequest getClearingRequestByProjectId(String projectId, User sw360User) throws TException {
-        return getThriftModerationClient().getClearingRequestByProjectId(projectId, sw360User);
+        try {
+            return getThriftModerationClient().getClearingRequestByProjectId(projectId, sw360User);
+        } catch (SW360Exception sw360Exp) {
+            if (sw360Exp.getErrorCode() == 404) {
+                throw new ResourceNotFoundException("Requested ClearingRequest not found");
+            } else if (sw360Exp.getErrorCode() == 403) {
+                throw new AccessDeniedException(
+                        "ClearingRequest or its Linked Project are restricted and / or not accessible");
+            } else {
+                log.error("Error fetching clearing request by project id: " + sw360Exp.getMessage());
+                throw sw360Exp;
+            }
+        }
     }
 
     public ClearingRequest getClearingRequestById(String id, User sw360User) throws TException {
-        return getThriftModerationClient().getClearingRequestById(id, sw360User);
+        try {
+            return getThriftModerationClient().getClearingRequestById(id, sw360User);
+        } catch (SW360Exception sw360Exp) {
+            if (sw360Exp.getErrorCode() == 404) {
+                throw new ResourceNotFoundException("Requested ClearingRequest not found");
+            } else if (sw360Exp.getErrorCode() == 403) {
+                throw new AccessDeniedException(
+                        "ClearingRequest or its Linked Project are restricted and / or not accessible");
+            } else {
+                log.error("Error fetching clearing request by id: " + sw360Exp.getMessage());
+                throw sw360Exp;
+            }
+        }
+    }
+
+    public Set<ClearingRequest> getMyClearingRequests(User sw360User, ClearingRequestState state) throws TException {
+        Set<ClearingRequest> clearingrequests = new HashSet<>(getThriftModerationClient().getMyClearingRequests(sw360User));
```

Comment:
this will only fetch the `Clearing Requests` where a user is either `Requesting User` or `Clearing Team`.
Please fetch all the CR based on user `Business Unit` so that CR fetched from REST API is in sync with CR shown in UI.